### ISSUE TITLE
Adapt to AVX2

### ISSUE THREAD

**Issue by torognes** [torognes/swarm#12]
Adapt SWARM to [AVX2](https://en.wikipedia.org/wiki/Advanced_Vector_Extensions#Advanced_Vector_Extensions_2) and the 256-bit registers available in the new Intel Haswell CPUs that became available in June 2013. Should allow 32-way SIMD parallellisation.

**Comment by torognes** [comment]
Or use https://github.com/RonnySoak/libssa

**Comment by colinbrislawn** [comment]
Parasail implements SIMD parallelisation, including using AVX2.
https://github.com/jeffdaily/parasail/ 

It's also very fast
https://github.com/jeffdaily/parasail/blob/2967c065de02fc2dc7e44050b4fc6ecfa064b2dc/performance.md

**Comment by frederic-mahe** [comment]
Thanks @colinbrislawn, I didn't know about [parasail](https://github.com/jeffdaily/parasail/).

**Comment by torognes** [comment]
Parasail looks interesting and flexible. A master student of mine have been working on another solution which I think is generally faster: https://github.com/RonnySoak/libssa
His master thesis will soon be available. We hope to integrate his code in Swarm and VSEARCH.

**Comment by frederic-mahe** [comment]
[Edlib](https://doi.org/10.1093/bioinformatics/btw753) is a very fast Levenshtein distance library, suitable for long sequences (> 1 Mb).

**Comment by frederic-mahe** [comment]
What are we going to do about this issue?

> Considering the relatively low number of computers in use with AVX2 and the rather limited time now spend on computing alignments, this performance improvement with AVX2 is currently rather limited.

Now that AVX2 capable CPUs are more frequent, do you think it would be interesting to implement this? It is difficult for me to access the amount of work necessary, but I imagine that work could be re-used to speed-up vsearch's search function.

**Comment by frederic-mahe** [comment]
Recent [paper](https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btad151/7085594) on pairwise alignment. This paper introduces [Scrooge](https://github.com/CMU-SAFARI/Scrooge) and compares it to [Edlib](https://doi.org/10.1093/bioinformatics/btw753) and to [GenASM](https://github.com/CMU-SAFARI/GenASM).

**Comment by frederic-mahe** [comment]
Indeed, [Wavefront](https://academic.oup.com/bioinformatics/article/37/4/456/5904262) is a very interesting alternative. There is even a GPU version now: [WFA-GPU](https://github.com/quim0/WFA-GPU) ([Aguado-Puig et al. 2023](https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btad701/7425447)).
